client/src/components/AttributeTrackFilters.vue

```diff
@@ -0,0 +1,294 @@
+import AttributeTrackFilterVue from '../../dive-common/components/AttributeTrackFilter.vue';
+import { useAttributes, useTrackFilters, useTrackStyleManager } from '../provides';
+
+
+export default defineComponent({
+  name: 'AttributeTrackFilters',
+
+  props: {
+  },
+
+  components: {
+    AttributeTrackFilterVue,
+  },
+
+  setup() {
+    const trackFilters = useTrackFilters();
+    const attributes = useAttributes();
+    const typeStylingRef = useTrackStyleManager().typeStyling;
+    const attributeTypes = computed(() => {
+      const typeMap: Record<string, string> = {};
+      attributes.value.forEach((item) => {
+        typeMap[item.name] = item.datatype;
+      });
+      return typeMap;
+    });
+
+    const editingFilter: Ref<null | number> = ref(null);
+    const addFilterDialog = ref(false);
+    const filters = trackFilters.attributeFilters;
+
+    // editing/adding Attribute Filter
+    const editingAtrKey = ref('');
+    const editName = ref('');
+    const editingOps = ref(['=', '!=', '>', '<', '>=', '<=', 'range', 'in', 'rangeFilter']);
+    const editingAtrOp: Ref<MatchOperator> = ref('=');
+    const editingAtrVal: Ref<string[] | string | number | number[] | null | boolean> = ref('');
+    const editingRange: Ref<number[]> = ref([0, 1]);
+    const editingAtrTypeList = ref(['track', 'detection']);
+    const editingFilterEnabled = ref(false);
+    const editingFilterDisplay = ref(false);
+    const editingAtrType: Ref<'track' | 'detection'> = ref('track');
+    const editingUserDefined = ref(true);
+    const attributeList = computed(() => attributes.value.filter((item) => item.belongs === editingAtrType.value).map((item) => item.name));
+
+    const changeAttributeType = () => {
+      if (editingAtrOp.value === 'range') {
+        editingAtrVal.value = [0, 1];
+      } else if (editingAtrOp.value === 'rangeFilter') {
+        editingAtrVal.value = 0;
+      } else {
+        editingAtrVal.value = '';
+      }
+    };
+
+    const addEditTrackFilter = (index?: number) => {
+      if (index !== undefined) {
+        const filter = filters.value[0];
+        editingAtrKey.value = filter.attribute;
+        editName.value = filter.name;
+        editingFilterEnabled.value = filter.enabled;
+        editingAtrType.value = filter.type;
+        editingAtrOp.value = filter.filter.op;
+        editingAtrVal.value = filter.filter.val;
+        editingUserDefined.value = filter.filter.userDefined || false;
+        if (filter.filter.range) {
+          editingRange.value = filter.filter.range;
+        }
+        editingFilter.value = index;
+      } else {
+        editingFilter.value = filters.value.length;
+      }
+      addFilterDialog.value = true;
+    };
+
+    const cancelFilter = () => {
+      editingAtrKey.value = '';
+      editingAtrOp.value = '=';
+      editingAtrVal.value = '';
+      addFilterDialog.value = false;
+    };
+
```

Comment:
Maybe re-name to 'resetFilter' or 'resetFilterFields' I think this might be used in other places in the future.